File: leetcode/csharp/1.two-sum.cs
using System.Collections.Generic;

public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> dict = new Dictionary<int,int>();
        for (int i = 0; i < nums.Length; i++)
        {
            if (dict.ContainsKey(target - nums[i]))
                return new int[] { i, dict[target - nums[i]] };
            else
                {
                    /// C#不允许重复add已经存在的键
                    if(!dict.ContainsKey(nums[i])){
                        dict.Add(nums[i],i);
                    }
                }
        }
        return new int[]{0,0};
    }
}

File: leetcode/kt/1.two-sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val map = TreeMap<Int, Int>()
        for (i in nums.indices) {
            if (map.containsKey(target - nums[i])) {
                println(map[target - nums[i]])
                return intArrayOf(map[target - nums[i]] ?: 0, i)
            } else {
                map[nums[i]] = i
            }
        }
        return intArrayOf(0, 0)
    }
}
